src/afrolabs/components/kafka.clj

```diff
@@ -656,14 +774,16 @@
       (post-init-hook
           [_ consumer]
         (let [assignment
-              (loop []
+              (loop [remaining 2]
+                (when (zero? remaining)
+                  (throw (ex-info "SeekToTimestampOffset Cannot find topic assignment to seek to" {:remaining remaining})))
```

Comment:
adding a guard to a condition we've not encountered yet. It is possible that this consumer may not receive a topic assignment at this stage. If that should happen with the old behaviour, the consumer will never stop initializing, and get stuck in the component init function as it is trying to seek to an offset on partitions that are not getting assigned.. If that happens, then all bets are off anyway, so an exception here is better outcome than getting stuck in a loop.